### PR TITLE
add dynamic-rule form demo

### PR DESCRIPTION
**Pull request by ddcat1115**
Meet consultation for several times

### CODE REVIEW

File: components/form/demo/dynamic-rule.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    this.setState({
+      checkNick: e.target.checked,
+    }, () => {
+      this.props.form.validateFields(['nickname'], { force: true });
+    });
+  }
+  render() {
+    const { getFieldDecorator } = this.props.form;
+    return (
+      <div>
```
Comment (benjycui) [inline components/form/demo/dynamic-rule.md]: `div` => `Form`

**Comment by benjycui** [comment]
优秀 👍

**Comment by afc163** [comment]
@ddcat1115

**Comment by benjycui** [comment]
ref: a9a8f68

**Comment by lili21** [comment]
https://codesandbox.io/s/jvnrlnql53

weird for this one.

**Pull request merged**